ui/src/components/workflow/WorkflowMetaDetails.js

```diff
@@ -74,4 +73,55 @@ class WorkflowMetaDetails extends Component {
     );
   }
 };
-export default connect(state => state.workflow)(WorkflowMetaDetails);
+
+class JSONTab extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      editingJSON: false,
+      wfs: this.props.wfs,
+      isNotParsable: false
+    }
+    this.editJSONswitch = this.editJSONswitch.bind(this);
+  }
+
+  editJSONswitch(e, which) {
```

Comment:
We should parse the edited JSON to parent component and then re-render the input form with new edited data. Could be done e.g. via passing reference to parent method **getWorkflowInputDetails()** and listen for event, or even some better way.